includes/class-fs-logger.php

```diff
@@ -691,5 +695,38 @@ public static function get_logs_download_url( $filename = '' ) {
 			return rtrim( $upload_dir['url'], '/' ) . $filename;
 		}
 
+		/**
+		 * @param $file_path
+		 * @param $query_results
+		 *
+		 * @return bool
+		 */
+		private static function write_csv_to_filesystem( $file_path, $query_results ) {
+			require_once( ABSPATH . 'wp-admin/includes/file.php' );
```

Comment:
@DanieleAlessandra I think it should be like this:
```
if ( empty( $query_results ) ) {
    return false;
}

require_once( ABSPATH . 'wp-admin/includes/file.php' );

WP_Filesystem();
global $wp_filesystem;

$headers = array_keys( (array) $query_results[0] );
```

This way, we won't trigger the other lines if we will return `false` anyway.

Also, the `require_once` line doesn't appear to be needed, as we are in a WordPress admin context where that is already included.